public/scripts/index.js

```diff
@@ -1,138 +1,162 @@
+
+let currentFrame
+let intermediateFrame // For holding the neighbour count
+let nextFrame
+const context = canvas.getContext('2d')
+const cellSize = 8
+let status
+const value = 0
+window.addEventListener('DOMContentLoaded', init)
+window.addEventListener('resize', resize)
+function init () {
+  console.log('INIT')
+  const simulateBtn = document.getElementById('simulate-button')
+  simulateBtn.addEventListener('click', simulate)
+  const clearBtn1 = document.getElementById('clear-button')
+  clearBtn1.addEventListener('click', clear)
+  const drawPresetBtn = document.getElementById('drawPreset-button')
+  drawPresetBtn.addEventListener('click', drawPreset)
+  const animateBtn = document.getElementById('animate')
+  animateBtn.addEventListener('click', animate)
+  const body = document.getElementById('body').addEventListener('click', handleMouseClick)
+  resize()
 }
-function handleMouseClick(event){
-    let spans = init();
-    const boundingRect = spans.canvas.getBoundingClientRect();
-    let point = calcClickCoordinates(event.clientX, event.clientY, boundingRect.left, boundingRect.top, boundingRect.right, boundingRect.bottom);
-    if(point){
-        let point2 = calcCellCoordinates(point.x , point.y, cellSize);
-        if(currentFrame[point2.y][point2.x] === 0){
-            currentFrame[point2.y][point2.x] = 1;
-        }
-        else{
-            currentFrame[point2.y][point2.x] = 0;
-        }
-        draw(currentFrame, context, cellSize);
-        spans.spanMouseX.innerHTML = 'Column ' + point2.x;
-        spans.spanMouseY.innerHTML = 'Row ' + point2.y;
-    }
-    else {
-        spans.spanMouseX.innerHTML = 'Out of bounds';
-        spans.spanMouseY.innerHTML = 'Out of bounds';
+function handleMouseClick (event) {
+  const spanMouseX = document.getElementById('mouse-x')
+  const spanMouseY = document.getElementById('mouse-y')
+  const canvas = document.getElementById('canvas')
+  const boundingRect = canvas.getBoundingClientRect()
+  const point = calcClickCoordinates(event.clientX, event.clientY, boundingRect.left, boundingRect.top, boundingRect.right, boundingRect.bottom)
+  if (point) {
+    const point2 = calcCellCoordinates(point.x, point.y, cellSize)
+    if (currentFrame[point2.y][point2.x] === 0) {
+      currentFrame[point2.y][point2.x] = 1
+    } else {
+      currentFrame[point2.y][point2.x] = 0
     }
+    draw(currentFrame, context, cellSize)
+    spanMouseX.innerHTML = 'Column' + point2.x
+    spanMouseY.innerHTML = 'Row ' + point2.y
+  } else {
+    spanMouseX.innerHTML = 'Out of bounds'
+    spanMouseY.innerHTML = 'Out of bounds'
+  }
 }
-function getInputValue() {
-    let rows = document.getElementById("rows").value;
-    let cols = document.getElementById("columns").value;
+function getInputValue () {
+  /* const rows = document.getElementById('rows').value
```

Comment:
Since the canvas size is now dynamic, we don't need this func, so just delete it. Commenting blocks of code doesn't make sense when working with git because it remembers all changes, so you can always revert.